ui/src/app/pages/analysis/leonardo-app-launcher.spec.tsx

```diff
@@ -136,63 +167,41 @@ describe('NotebookLauncher', () => {
   });
 
   it('should render', async () => {
-    const wrapper = await notebookComponent();
-    expect(wrapper).toBeTruthy();
+    notebookComponentAlt();
+    expect(
+      screen.getByRole('heading', {
+        name: /loading notebook: wharrgarbl/i,
+      })
+    ).toBeInTheDocument();
   });
 
   it('should show redirect display before showing notebook', async () => {
-    const wrapper = await notebookComponent();
-    expect(wrapper.exists('[data-test-id="leo-app-launcher"]')).toBeTruthy();
+    notebookComponentAlt();
+    expect(screen.getByTestId('leo-app-launcher')).toBeInTheDocument();
   });
 
   it('should be "Initializing" until a Creating runtime for an existing notebook is running', async () => {
     runtimeStub.runtime.status = RuntimeStatus.CREATING;
 
-    const wrapper = await notebookComponent();
-    await waitForFakeTimersAndUpdate(wrapper);
-
-    expect(
-      wrapper.exists(
-        getCardSpinnerTestId(ProgressCardState.UnknownInitializingResuming)
-      )
-    ).toBeTruthy();
-    expect(currentCardText(wrapper)).toContain(
-      notebookProgressStrings.get(Progress.Initializing)
-    );
-
-    runtimeStub.runtime.status = RuntimeStatus.RUNNING;
-    await waitForFakeTimersAndUpdate(wrapper);
+    notebookComponentAlt();
 
-    expect(
-      wrapper.exists(getCardSpinnerTestId(ProgressCardState.Redirecting))
-    ).toBeTruthy();
-    expect(currentCardText(wrapper)).toContain(
-      notebookProgressStrings.get(Progress.Redirecting)
+    await screen.findByTestId(
+      getCardSpinnerTestIdAlt(ProgressCardState.UnknownInitializingResuming)
     );
-  });
 
-  it('should be "Initializing" until a Creating runtime for a new notebook is running', async () => {
```

Comment:
This seems to be identical in all but name to the test above it, so I deleted it.

Reply:
The purpose of the difference in tests is clear (existing notebook vs new notebook) but I'm not sure if that can really be tested here.  hm.

Reply:
Looks like `history.push()` is the difference, according to tests below.  Are these similar enough to use a `test.each()` approach?  I'd like to restore this test using the `each()` pattern.

Reply:
Thanks! I brought it back, and it can be found in the updated code.